### PR TITLE
EclusiveExecutor utility

### PR DESCRIPTION
**Pull request by CheatEx**
Put this pattern into a common utility with an example refactoring

### CODE REVIEW

File: util/base/src/main/java/jetbrains/jetpad/base/ExclusiveExecutor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package jetbrains.jetpad.base;
```
Comment (glebleonov) [inline util/base/src/main/java/jetbrains/jetpad/base/ExclusiveExecutor.java]: Licence should be added

**Comment by glebleonov** [comment]
Why do we need to extract such an utility? Do you know any other places where ExclusiveExecutor can be used? As for me, the its behavior is rather tricky and easily can be misused.

**Comment by azlobinjb** [comment]
TokenListEditor#mySyncing
PyHybridProperties:89
PrettyHybridProperty#myInSync
DlCodeTextProperty#mySyncing
HalfWayPropertyBinding#mySyncing

There are more, just with another field name.

**Comment by CheatEx** [comment]
We come up with a better design

**Pull request closed without merging**